core/server/models/base/index.js

```diff
@@ -763,7 +777,16 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
         const id = options.id;
         const model = this.forge({id: id});
 
-        data = this.filterData(data);
+        if (options.context && !options.context.internal && data.id && data.id !== options.id) {
```

Comment:
PR title does not match what the PR is doing or?
It protects against id mismatch? Let me know :)

Reply:
@kirrg001 The PR is to stop users from setting the id of a resource, either when creating it, or when editing it. The prevention of ID mismatch is just the way in which we stop them setting the ID.

Reply:
https://github.com/TryGhost/Ghost/pull/9857#discussion_r215883320 from this conversation

Reply:
Did you see https://github.com/TryGhost/Ghost/blob/2.5.0/core/server/api/v0.1/utils.js#L427?

Reply:
I currently think both checks should live in the v2 API validation + serialization layer 🤔 

- editId comparison
- add should not contain id